### ISSUE TITLE
Refactor Loss Functions

### ISSUE THREAD

**Issue by philippmwirth** [lightly-ai/lightly#503]
# Refactor Loss Functions

Some of the loss functions are inherently symmetrized (e.g. `SymNegCosineSimilarityLoss`) while others need to be symmetrized manually (e.g. `NTXEnropyLoss` with memory bank). This issue is about figuring out if and how we can streamline the loss functions and implementing the necessary changes.

**Comment by philippmwirth** [comment]
The detaching of the features in the symmetric negative cosine similarity should be moved outside as mentioned in #559.

**Comment by guarin** [comment]
I propose to refactor the SymNegCosineSimLoss to the non-symmetric NegCosineSimilarity:
```
class NegCosineSimilarity(nn.Module):
	def forward(self, p, z):
		return -torch.nn.functional.cosine_similarity(p, z).mean()
```

Where `z` is assumed to be detached.

The we get the following loss calculation:
```
criterion = NegCosineSimilarity()
loss = 0.5 * (criterion(p0, z0) + criterion(p1, z1))
```

**Comment by philippmwirth** [comment]
Sounds good! I would call it `NegativeCosineSimilarity` because `Neg` is an unnecessary abbreviation.
